etcdserver/server.go

```diff
@@ -738,19 +742,18 @@ func (s *EtcdServer) run() {
 		case leases := <-expiredLeaseC:
 			s.goAttach(func() {
 				// Increases throughput of expired leases deletion process through parallelization
-				c := make(chan struct{}, maxPendingRevokes)
 				for _, lease := range leases {
 					select {
-					case c <- struct{}{}:
 					case <-s.stopping:
 						return
+					default:
 					}
+					revokerLimiter.Wait(context.Background())
```

Comment:
this rate limiting probably belongs in `lessor.runLoop` instead of the etcd server